strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/AuthorizationConfigController.java

```diff
@@ -36,109 +48,90 @@
 
     @Inject
-    UserService userService;
+    private UserService userService;
 
     @Inject
-    CacheManager cacheManager;
+    private CacheManager cacheManager;
 
     @Inject
-    AnonymousAuthenticationFilter anonymousAuthenticationFilter;
+    private AnonymousAuthenticationFilter anonymousAuthenticationFilter;
 
-    private AuthorizationConfig config;
+    @Inject
+    private ConversionService conversionService;
 
-    private ResponseEntity processConfig(Consumer<AuthorizationConfig> consumer)
-    {
-        return processConfig(consumer, config -> ResponseEntity.ok()
-                                                               .build());
-    }
+    // ----------------------------------------------------------------------------------------------------------------
+    // Add role
 
-    private ResponseEntity processConfig(Consumer<AuthorizationConfig> consumer,
-                                         CustomSuccessResponseBuilder customSuccessResponseBuilder)
+    @ApiOperation(value = "Used to add new roles")
+    @ApiResponses(value = { @ApiResponse(code = 200,
+                                         message = SUCCESSFUL_ADD_ROLE),
+                            @ApiResponse(code = 400,
+                                         message = FAILED_ADD_ROLE) })
+    @PostMapping(value = "/role",
+                 consumes = MediaType.APPLICATION_JSON_VALUE,
+                 produces = { MediaType.TEXT_PLAIN_VALUE,
+                              MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity addRole(@RequestBody @Validated RoleForm roleForm,
+                                  BindingResult bindingResult,
+                                  @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
-        Optional<AuthorizationConfig> configOptional = configProvider.getConfig();
-
-        if (configOptional.isPresent())
+        if (roleForm == null)
         {
-            try
-            {
-                config = configOptional.get();
-
-                if (consumer != null)
-                {
-                    consumer.accept(config);
-                }
-
-                return customSuccessResponseBuilder.build(config);
-            }
-            catch (Exception e)
-            {
-                logger.error("Error during config processing.", e);
-                return toError("Error during config processing: " + e.getLocalizedMessage());
-            }
+            throw new RequestBodyValidationException("Empty request body", bindingResult);
         }
-        else
+        if (bindingResult.hasErrors())
         {
-            return toError("Unable to locate AuthorizationConfig to update...");
+            throw new RequestBodyValidationException(FAILED_ADD_ROLE, bindingResult);
         }
-    }
 
-    // ----------------------------------------------------------------------------------------------------------------
-    // Add role
+        // Convert RoleForm to Role
```

Comment:
You don't have to add comments like this, followed line is self explained enough